packages/lib-webpack/src/webpack/DynamicRemotePlugin.ts

```diff
@@ -0,0 +1,167 @@
+import {
+  extensionArraySchema,
+  remoteEntryScript,
+  remoteEntryCallback,
+} from '@openshift/dynamic-plugin-sdk';
+import type { Extension } from '@openshift/dynamic-plugin-sdk';
+import * as _ from 'lodash-es';
+import * as fs from 'fs';
+import * as glob from 'glob';
+import * as path from 'path';
+import * as webpack from 'webpack';
+import type { PluginBuildMetadata } from '../types/plugin';
+import { pluginBuildMetadataSchema } from '../yup-schemas';
+import { GenerateManifestPlugin } from './GenerateManifestPlugin';
+import { PatchContainerEntryPlugin } from './PatchContainerEntryPlugin';
+
+const parseJSONFile = (filePath: string) => JSON.parse(fs.readFileSync(filePath, 'utf-8'));
+
+const loadPluginMetadata = (fileName: string, baseDir = process.cwd()) => {
+  const filePath = path.resolve(baseDir, fileName);
+  return validatePluginMetadata(parseJSONFile(filePath));
```

Comment:
I agree... we want to validate it in both cases, doing a load then validate in one is not ideal imo.

```
// calling code
if ('string')
  // load file and validate contents
else
  // validate contents
```
Why not just combine them... if the contents are invalid we can handle that error together rather than separate (if later desired). Loading should just load.
```
// calling code
if ('string') {
  // load file
}
// validate contents
```
It's the same content, just two different delivery mechanisms... for maintenance I think we should handle those two together.

Reply:
Agreed, it's better to separate load vs. validation logic, will do the change.